pkg/controller/perconaservermongodb/connections.go

```diff
@@ -19,11 +20,42 @@ type MongoClientProvider interface {
 
 func (r *ReconcilePerconaServerMongoDB) MongoClientProvider() MongoClientProvider {
```

Comment:
Since `MongoClientProvider` is now only used in the package, we can convert to non-exported


PS: we need to change the name to something like getMongoClientProvider since we have the type with name conflict.

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1828/commits/fa35daf13176700011c72cf58271a20fb9259112